### PR TITLE
Remove `publish` target from artman and config

### PR DESCRIPTION
**Pull request by andreamlin**
Remove (github and local) publishing artman configs, as these are not needed by ACTools or Yoshi team.

This reduces the size of the artman configs.

The old `artman.tasks.publish.noop` file is renamed as `artman.tasks.emit_success`, which more accurately describes that task.

In the CircleCI smoke tests, change all `publish` targets to `generate`.

Eventually, we would be able to remove the `generate` target because it's the only option, and it would be frivolous to specify it if there are no other options.

### CODE REVIEW

**Comment by andreamlin** [comment]
Waiting for smoke tests to pass before sending out for review.

**Comment by andreamlin** [comment]
PTAL

**Comment by andreamlin** [comment]
1. Because googleapis is synced from internal repos, wouldn't the changes get overridden by internal pushes?

2. Sure.

**Comment by garrettjonesgoogle** [comment]
1. Of course - the change would need to be done in the internal files (I thought that was implied...)

**Comment by andreamlin** [comment]
1. Any changes made to the internal files would get overridden as soon as an API producer regenerates or bootstraps a config using a version of artman without this PR.

**Comment by garrettjonesgoogle** [comment]
1. API producers won't regenerate artman files. In the case of config bootstraps, I think we can handle the 0-2 cases that might happen in the "gap" on a one-off basis.

**Comment by andreamlin** [comment]
Ok, made an internal PR for removing all instances of the `publish_targets`.
Also made an internal PR for deprecating g3artman.

**Comment by garrettjonesgoogle** [comment]
I think you mean a PR to deprecate the *publish feature* in g3artman?

**Comment by andreamlin** [comment]
Ok, (1) and (2) are done!
PTAL

File: artman/config/proto/config.proto [hunk 1]
```diff
@@ -266,6 +252,7 @@ message Artifact {
   // [dependencies.yaml](https://github.com/googleapis/googleapis/blob/master/gapic/packaging/dependencies.yaml)
   PackageVersion package_version = 14;
 
+  // DEPRECATED.
```
Comment (garrettjonesgoogle) [inline artman/config/proto/config.proto]: Maybe use `UNSUPPORTED` (here and below)

File: artman/pipelines/code_generation.py [hunk 2]
```diff
@@ -23,6 +23,7 @@
 from artman.pipelines import pipeline_base
 from artman.tasks import io_tasks
 from taskflow.patterns import linear_flow
+from artman.tasks import emit_success
```
Comment (garrettjonesgoogle) [inline artman/pipelines/code_generation.py]: This doesn't appear to be referenced in this file

File: artman/pipelines/gapic_generation.py [hunk 3]
```diff
@@ -18,14 +18,15 @@
...
 from artman.pipelines import code_generation as code_gen
 from artman.pipelines import grpc_generation as grpc_gen
+from artman.tasks import emit_success
 from artman import tasks
 from artman.utils import task_utils
 
 
 # kwargs required by GAPIC code gen
-_GAPIC_REQUIRED = ['service_yaml', 'gapic_yaml', 'language', 'aspect', 'publish']
+_GAPIC_REQUIRED = ['service_yaml', 'gapic_yaml', 'language', 'aspect',]
```
Comment (garrettjonesgoogle) [inline artman/pipelines/gapic_generation.py]: remove the trailing comma

Comment (andreamlin) [reply artman/pipelines/gapic_generation.py]: Done.

File: artman/pipelines/code_generation.py (hunk 2 above)
Comment (andreamlin) [reply artman/pipelines/code_generation.py]: Done.

File: artman/config/proto/config.proto (hunk 1 above)
Comment (andreamlin) [reply artman/config/proto/config.proto]: Done.

**Pull request merged**